echopop/extensions/diagnostics.py

```diff
@@ -0,0 +1,510 @@
+                        }
+                    )
+                }
+            )
+        # ---- Transect
+        if "transect" in self.data.results:
+            strat_krig = self.data.results["stratified"]["transect"].copy()
+            # ---- Add 'biomass'
+            DATAINPUT["Biomass"].update(
+                {
+                    "transect": pd.DataFrame(
+                        {
+                            "stratum": strata_info.stratum_inpfc,
+                            "total": strat_krig["estimate"]["strata"]["total"] * 1e-6,
+                            "lower": [k[0] * 1e-6 for k in strat_krig["ci"]["strata"]["total"]],
+                            "upper": [k[1] * 1e-6 for k in strat_krig["ci"]["strata"]["total"]],
+                        }
+                    )
+                }
+            )
+            # ---- Add 'biomass density'
+            DATAINPUT["Biomass density"].update(
+                {
+                    "transect": pd.DataFrame(
+                        {
+                            "stratum": strata_info.stratum_inpfc,
+                            "density": strat_krig["estimate"]["strata"]["density"] * 1e-6,
+                            "lower": [k[0] * 1e-6 for k in strat_krig["ci"]["strata"]["density"]],
+                            "upper": [k[1] * 1e-6 for k in strat_krig["ci"]["strata"]["density"]],
+                        }
+                    )
+                }
+            )
+
+        # Initialize `panel` extension
+        pn.extension()
+
+        # Initialize widgets
+        # ---- Variable options
+        var_options = ["Biomass", "Biomass density"]
+        # ---- Get column names
+        var_columns = {"Biomass": "total", "Biomass density": "density"}
+        # ---- y-axis names
+        y_axis_label = {
+            "Biomass": "Biomass (kmt) [mean ± 95% CI]",
+            "Biomass density": "Biomass density (kmt nmi⁻²) [mean ± 95% CI]",
+        }
+        # ---- Create dropdown selector
+        var_dropdown = pn.widgets.Select(name="Variable", options=var_options)
+
+        # Create plotting function
+        @pn.depends(selection=var_dropdown)
+        def update_plot(selection):
+
+            # Parse dataset
+            data = DATAINPUT[selection]
+            # ---- Get data column
+            value = var_columns[selection]
+
+            # Initialize plotting layers
+            plot_layers = []
+
+            # Plot the data
+            # ---- Kriging
+            if "kriging" in data:
+                # ---- Add errorbar
+                kriging_pt_err = hv.Segments(
+                    data["kriging"], kdims=["stratum", "lower", "stratum", "upper"], label="Kriging"
+                ).opts(color="red")
+                # ---- Point estimates
+                kriging_pts = hv.Points(
+                    data["kriging"], kdims=["stratum", value], label="Kriging"
+                ).opts(
```

Comment:
I suggest slightly offsetting the black and red points horizontally so that error bars can be easily seen and compared. It is hard to see right now with the overlap, like what's shown in the example fig.